llvm/lib/CheerpUtils/PHIHandler.cpp

```diff
@@ -251,30 +232,30 @@ void EndOfBlockPHIHandler::runOnEdge(const Registerize& registerize, const Basic
 				}
 			}
 		}
-		phiRegs.insert(std::make_pair(phiReg, PHIRegData(phi, std::move(incomingRegisters), selfReferencing)));
 	}
 
 	for (auto& X : phiRegs)
 	{
+		std::sort(X.second.incomingRegs.begin(), X.second.incomingRegs.end(), [](auto&& a, auto&& b) { return a.first < b.first; });
```

Comment:
Could you confirm that sorting was required here? I plan to merge the PR anyway, but I am curious

Reply:
it is required by an assertion later. if **that** is necessary, not sure.